source/utilities.cc

```diff
@@ -745,3 +745,65 @@ namespace aspect
     template std_cxx1x::array<double,3> spherical_coordinates<3>(const Point<3> &position);
   }
 }
+
+namespace dealii
+{
+  namespace Utilities
+  {
+    namespace MPI
+    {
+      template<int dim>
+      Tensor<1,dim> 
+      reduce_tensor(const Tensor<1,dim> &local, const MPI_Comm &mpi_communicator)
+      {
+        std::vector<double> entries(dim);
+        std::vector<double> global_entries(dim);
+        Tensor<1,dim> global;
+
+        for(unsigned int i=0; i<dim; ++i)
+          entries[i] = local[i];
+
+        MPI_Allreduce (&entries[0], &global_entries[0], dim, MPI_DOUBLE,
```

Comment:
I think there's actually a `Utilities::MPI::sum` function for arrays to which you can just defer here.